### PR TITLE
Convert Spring Scheduling groovy tests to java

### PR DESCRIPTION
**Pull request by vallabhnatu**
Hi,

This PR is related to #7195. This converts spring scheduler groovy tests to java. Additionally, a TO-DO of @anuraaga has been completed in `InstrumentationTestRunner.java`.

### CODE REVIEW

File: instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
+/*
```
Comment (laurit) [inline instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java:1]: usually we place java tests in the same package as the instrumentation code, which here would be `io.opentelemetry.javaagent.instrumentation.spring.scheduling.v3_1`

File: instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    try (AnnotationConfigApplicationContext context =
+        new AnnotationConfigApplicationContext(TriggerTaskConfig.class)) {
+      TriggerTask task = context.getBean(TriggerTask.class);
+      task.blockUntilExecute();
+
+      assertThat(task).isNotNull();
+      testing.waitAndAssertTraces(
+          trace ->
+              trace
+                  .hasSize(1)
```
Comment (laurit) [inline instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java:60]: usually we don't use `hasSize` as `hasSpansSatisfyingExactly` also verifies the size

File: instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java (hunk 1 above)
Comment (vallabhnatu) [reply instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java]: Moved tests to `io.opentelemetry.javaagent.instrumentation.spring.scheduling.v3_1` package

File: instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java (hunk 2 above)
Comment (vallabhnatu) [reply instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java]: Removed superfluous `hasSize` assertion

**Comment by vallabhnatu** [comment]
Thanks @laurit for the review. I have made the suggested changes.

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/InstrumentationTestRunner.java [hunk 3]
```diff
@@ -119,15 +121,29 @@ private <T extends Consumer<TraceAssert>> void waitAndAssertTraces(
     List<T> assertionsList = new ArrayList<>();
     assertions.forEach(assertionsList::add);
 
+    AtomicReference<AssertionError> assertionError = new AtomicReference<>();
     try {
       await()
-          .untilAsserted(() -> doAssertTraces(traceComparator, assertionsList, verifyScopeVersion));
+          .untilAsserted(
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/InstrumentationTestRunner.java:127]: Sorry for drive-by but as I was mentioned :) I don't remember details but do remember test failures being hard to debug before adding this hack with the bad traces, maybe there were some tests that had or have multiple assertions but again can't remember. I do remember being embarrassed to introduce awaitilitiy given the poor stack traces.

If the general test structure has changed or this corner case has been reduced enough it could still be fine, but want to reinforce the point that the stack traces can be important.

[the remark this answers is not in the document]
Comment (laurit) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/InstrumentationTestRunner.java:127]: Actually after these changes the behavior seems to be the same as with just using
`await().untilAsserted(() -> doAssertTraces(traceComparator, assertionsList, verifyScopeVersion));`
Junit5 seems to unwrap the `ConditionTimeoutException` and just shows the `AssertionFailedError` as the failure.
I think it is best not to make this change as it doesn't really resolve the underlying issue, which is to provide a stack trace that would let one easily determine which test failed. Since @anuraaga worked on this one thing that has changed is that we added `showStandardStreams = true` to `testLogging` to ease debugging some test failures. Because of that now when there is a test failure in CI it is easier to open the build scan rather than eyeball the test output and find the stack trace because there is a lot of output now. My guess would be that not having the full stack trace made it hard to quickly figure out which tests failed when you just looked at the test output.
To me the most annoying part of using awaitility `untilAsserted` here is that it really wants to make the tests pass. If there is a failure then it will try running the same asserts until the timeout. When there is a bug that makes a bunch of tests fail it takes a long time for the tests to complete. It might be better to change it so that it first waits for the requested amount of traces and then asserts on them once like the groovy tests do.

**Review by laurit — APPROVED** [review]

**Comment by vallabhnatu** [comment]
Thanks @laurit for merging the PR and sorry for the delay.
I wanted to see if it was feasible to run tests without awaitility but couldn't find enough time to do so.
